Applications/Spire/Include/Spire/Charting/Charting.hpp

```diff
@@ -15,11 +16,106 @@ namespace Spire {
   class DropDownColorPicker;
   class StyleDropDownMenuItem;
   class StyleDropDownMenuList;
+  struct TrendLine;
   class TrendLineEditor;
   class TrendLineModel;
   class TrendLineStyleDropDownMenu;
   using Candlestick =
     Nexus::TechnicalAnalysis::Candlestick<ChartValue, ChartValue>;
+
+  //! Returns the slope of the line reprsented by the given points.
+  /*!
+    \param x1 The x-coordinate of the first point.
+    \param y1 The y-cooridnate of the first point.
+    \param x2 The x-coordinate of the second point.
+    \param y2 The y-coordinate of the second point.
+  */
+  template <typename T>
+  T slope(T x1, T y1, T x2, T y2) {
```

Comment:
Use a ```double``` for this instead of a template for all of these functions. Instead of passing in x and y values, look into using ```QPointF```.

Reply:
I can make these functions take doubles, but I'll have to convert ChartPoints to QPointFs in TrendLineModel if I use QPointFs as parameters.

Reply:
Okay, use ```QPointF```s then.

Reply:
Should I create an alias for: static_cast<double>(static_cast<ChartValue>(value))?

Reply:
What's an alias?

Reply:
I don't remember the exact syntax, but isn't it possible to do something like:

using point_cast = static_cast<double>(static_cast<ChartValue>());

Reply:
No, use a function.